### PR TITLE
Native nuget

### PR DESCRIPTION
**Pull request by Or-Gabay**

### CODE REVIEW

File: artifactory/commands/nuget/nugetconfig.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	var vConfig *viper.Viper
+	configResult := &ConfigFile{}
+	configResult.Version = prompt.BUILD_CONF_VERSION
+	configResult.ConfigType = confType.String()
+	configResult.Resolver.ServerId, vConfig, err = prompt.ReadServerId()
+	if err != nil {
+		return err
+	}
```
Comment (yahavi) [inline artifactory/commands/nuget/nugetconfig.go]: This code is repeated at all native commands. Please use a helper function or use inheritance.

File: artifactory/commands/nuget/nugetconfig.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	err = ioutil.WriteFile(configFilePath, resBytes, 0644)
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	log.Info(confType.String() + " build config successfully created.")
+	return nil
```
Comment (yahavi) [inline artifactory/commands/nuget/nugetconfig.go]: This code is repeated at all native commands. Please use a helper function or use inheritance.

File: docs/artifactory/nugetconfig/help.go [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package nugetconfig
+
+const Description = "Generate nuget install configuration."
```
Comment (yahavi) [inline docs/artifactory/nugetconfig/help.go]:
```suggestion
const Description = "Generate nuget restore configuration."
```

File: artifactory/cli.go [hunk 4]
```diff
@@ -1578,10 +1597,36 @@ func dockerPullCmd(c *cli.Context) error {
...
+		return err
+	}
+
+	if exists {
+		// Found a config file.
+		args, err := shellwords.Parse(strings.Join(extractCommand(c), " "))
+		if err != nil {
+			return errorutils.CheckError(err)
+		}
+		// Validates the go command. If a config file is found, the only flags that can be used are build-name, build-number and module.
```
[the remark this answers is not in the document]
Comment (Or-Gabay) [reply artifactory/cli.go]: Fixed.

File: docs/artifactory/nugetconfig/help.go (hunk 3 above)
Comment (Or-Gabay) [reply docs/artifactory/nugetconfig/help.go]: Fixed

File: artifactory/commands/nuget/nugetconfig.go (hunk 1 above)
Comment (Or-Gabay) [reply artifactory/commands/nuget/nugetconfig.go]: Fixed

File: artifactory/commands/nuget/nugetconfig.go (hunk 2 above)
Comment (Or-Gabay) [reply artifactory/commands/nuget/nugetconfig.go]: Fixed

**Pull request merged**